src/components/map/MapService.js

```diff
@@ -843,13 +854,12 @@ goog.require('ga_urlutils_service');
          * layers.
          */
         this.getBackgroundLayers = function() {
-          var self = this;
-          if (!layers) {
-            return;
-          }
+          var that = this;
           return $.map(gaTopic.get().backgroundLayers, function(bodId) {
-            var retVal = {id: bodId,
-                          label: self.getLayerProperty(bodId, 'label')};
+            var retVal = {
+              id: bodId,
+              label: that.getLayerProperty(bodId, 'label')
+            };
```

Comment:
This here is tricky. It depends on topic, but as well on gaLayers (for the getLayerProperty function). So it can be argued that it belongs here...